### PR TITLE
logger: Fix behavior outside of accumulate_means

### PR DESCRIPTION
**Pull request by shwang**
Calling `logger.record` outside of a `with logger.accumulate_means`
context should lead to a `record`, not a `record_means` call.

h/t @decodyng.

### CODE REVIEW

**Comment by shwang** [comment]
Something's breaking with jaxlib right now. I"ll give this a look

**Comment by decodyng** [comment]
@shwang Does the jaxlib issue you mentioned only impact imitation-specific tests, or does it also mean I should wait until it's fixed to merge it into the branch used for ILR?

**Comment by shwang** [comment]
Imitation-specific. Relevant Tests, lint, etc are passing on my local
machine.

**Comment by shwang** [comment]
Problem should be fixed by #265

File: tests/test_logger.py
Comment (shwang) [inline tests/test_logger.py:28]: Adding this line checks that the recorded "A": -1 is overwritten by  "A": 1 in the next line.

Previously, the observed value would be the mean of these two values (0) instead.

Comment (AdamGleave) [reply tests/test_logger.py:28]: This might be worth adding as a Python comment?

**Review by AdamGleave — APPROVED** [review]
LGTM

File: tests/test_logger.py
Comment (shwang) [reply tests/test_logger.py:28]: added

**Pull request merged**
